internal/reconciler/commonstatus/checker.go

```diff
@@ -0,0 +1,93 @@
+package commonstatus
+
+import (
+	"context"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+
+	"github.com/kyma-project/telemetry-manager/internal/conditions"
+	"github.com/kyma-project/telemetry-manager/internal/workloadstatus"
+)
+
+const (
+	SignalTypeTraces  = "traces"
+	SignalTypeMetrics = "metrics"
+	SignalTypeLogs    = "logs"
+)
+
+type DeploymentProber interface {
+	IsReady(ctx context.Context, name types.NamespacedName) error
+}
+
+type DaemonsetProber interface {
+	IsReady(ctx context.Context, name types.NamespacedName) error
+}
+
+type ErrorToMessageConverter interface {
```

Comment:
the same applies to this interface... it is defined multiple times. is there reason for this?